ufl/domain.py

```diff
@@ -410,26 +416,6 @@ def extract_domains(expr: Expr | Form, expand_mesh_sequence: bool = True):
         return sort_domains(join_domains(domainlist, expand_mesh_sequence=expand_mesh_sequence))
 
 
-def extract_unique_domain(expr, expand_mesh_sequence: bool = True):
-    """Return the single unique domain expression is defined on or throw an error.
-
-    Args:
-        expr: Expr or Form.
-        expand_mesh_sequence: If True, MeshSequence components are expanded.
-
-    Returns:
-        domain.
-
-    """
-    domains = extract_domains(expr, expand_mesh_sequence=expand_mesh_sequence)
-    if len(domains) == 1:
-        return domains[0]
-    elif domains:
-        raise ValueError("Found multiple domains, cannot return just one.")
-    else:
-        return None
-
-
```

Comment:
Can we not change the definition of `extract_unique_domain()` here? Instead, can we change
```
        for t in traverse_unique_terminals(expr):
            domainlist.extend(t.ufl_domains())
```
in `extract_domains()` function to use `DAGTraverser`? I think it is important to make sure that `extract_domains()` and `extract_unique_domains()` are consistent.